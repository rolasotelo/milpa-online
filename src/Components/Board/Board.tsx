import React from "react";
import { AnyCard } from "../../common/types";
import useGameContext from "../../Hooks/useGameContext/useGameContext";
import Calendar from "../Calendar/Calendar";
import Cards from "../Cards/Cards";
import { computeDisplayableBoards } from "../GameProvider/utils/helpers";
import Milpa from "../Milpa/Milpa";

interface Props {}

export type GoodsSlots = {
  top: (AnyCard | undefined)[];
  bottom: (AnyCard | undefined)[];
  left: (AnyCard | undefined)[];
  right: (AnyCard | undefined)[];
};

const Board = (props: Props) => {
  const context = useGameContext();
  const { leftBoard, rightBoard } = computeDisplayableBoards(context);

  return (
    <div className="flex flex-col">
      <Calendar />
      <div className="flex flex-row h-35rem ">
        <Milpa milpa={leftBoard.milpa} edges={leftBoard.edges} isYourMilpa />
        <Cards />
        <Milpa
          milpa={rightBoard.milpa}
          edges={rightBoard.edges}
          isYourMilpa={false}
        />
      </div>
    </div>
  );
};

export default Board;
