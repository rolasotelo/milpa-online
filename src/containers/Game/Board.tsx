import React from "react";
import Slot from "./Slot";
function EdgeTopOrBottom() {
    return (
        <div className="col-start-2 col-span-4 flex flex-row justify-between">
            <Slot />
            <Slot />
            <Slot />
            <Slot />
        </div>
    );
}

function EdgeMilpaEdge() {
    return (
        <>
            <div className="col-start-1 col-end-2">
                <Slot />
            </div>
            <div className="col-start-2 col-end-6 flex flex-row justify-between">
                <Slot />
                <Slot />
                <Slot />
                <Slot />
            </div>
            <div className="col-end-7 col-span-1 ...">
                <Slot />
            </div>
        </>
    );
}
export default function Board() {
    return (
        <div className="h-milpa-board-mobile w-milpa-board-mobile tablet:h-milpa tablet:w-milpa bg-yellow-800 m-3 p-1 tablet:p-2">
            <div className="grid grid-cols-6 gap-2 tablet:gap-4 place-content-center h-full">
                <EdgeTopOrBottom />
                <EdgeMilpaEdge />
                <EdgeMilpaEdge />
                <EdgeMilpaEdge />
                <EdgeMilpaEdge />
                <EdgeTopOrBottom />
            </div>
        </div>
    );
}