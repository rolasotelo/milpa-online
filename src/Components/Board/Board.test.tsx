import React from "react";
import renderer from "react-test-renderer";

import Board from "./Board";

describe("<Board />", () => {
  const defaultProps = {};
  const wrapper = renderer.create(<Board {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
