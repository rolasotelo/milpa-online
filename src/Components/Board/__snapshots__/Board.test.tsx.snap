// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> render 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="flex w-full h-16 bg-mexicanGreen-dark"
  >
    <div
      className="bg-mexicanGreen-light w-48 h-8"
    >
      ENERO
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      FEBRERO
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      MARZO
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      ABRIL
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      MAYO
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      JUNIO
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      JULIO
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      AGOSTO
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      SEPTIEMBRE
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      OCTUBRE
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      NOVIEMBRE
    </div>
    <div
      className="bg-mexicanBoneLight w-48 h-8"
    >
      DICIEMBRE
    </div>
  </div>
  <div
    className="flex flex-row h-35rem"
  >
    <div
      className="flex flex-col bg-mexicanGreen-light w-1/3"
    >
      <div
        className="py-4"
      >
        <div
          className="h-16 bg-mexicanBoneLight"
        >
          Your Turn
        </div>
      </div>
      <div
        className="m-4 p-6 bg-yellow-800 grid grid-cols-4 text-center gap-4"
      >
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥕
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌽
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥕
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🍅
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🍅
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🫑
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥕
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌽
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥔
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥕
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🪰
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌽
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌶
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥔
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🍅
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌽
        </div>
      </div>
      <div
        className="py-4"
      >
        <div
          className="h-16 bg-mexicanBoneLight"
        >
          100 cacao
        </div>
      </div>
    </div>
    <div
      className="flex flex-col bg-mexicanBlue w-1/3 p-10 justify-evenly"
    >
      <div
        className="bg-mexicanBone  h-28"
      >
        Corn
      </div>
      <div
        className="bg-mexicanBone  h-28"
      >
        Tomate
      </div>
      <div
        className="bg-mexicanBone  h-28"
      >
        Chile
      </div>
    </div>
    <div
      className="flex flex-col bg-mexicanGreen-light w-1/3"
    >
      <div
        className="py-4"
      >
        <div
          className="h-16 bg-mexicanBoneLight"
        >
          Your Turn
        </div>
      </div>
      <div
        className="m-4 p-6 bg-yellow-800 grid grid-cols-4 text-center gap-4"
      >
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥕
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌽
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥕
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🍅
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🍅
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🫑
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥕
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌽
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥔
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥕
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🪰
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌽
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌶
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🥔
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🍅
        </div>
        <div
          className="w-16 h-16 bg-yellow-900"
        >
          🌽
        </div>
      </div>
      <div
        className="py-4"
      >
        <div
          className="h-16 bg-mexicanBoneLight"
        >
          100 cacao
        </div>
      </div>
    </div>
  </div>
</div>
`;
